airbyte-webapp/src/packages/cloud/views/auth/Auth.tsx

```diff
@@ -54,7 +54,17 @@ export const Auth: React.FC = () => {
               <Route path={CloudRoutes.FirebaseAction} element={<FirebaseActionRoute />} />
               <Route
                 path="*"
-                element={<Navigate to={`${CloudRoutes.Login}${loggedOut ? "" : `?from=${pathname}`}`} />}
+                element={
+                  <Navigate
+                    to={
+                      user
+                        ? `${pathname}`
+                        : `${CloudRoutes.Login}${
+                            loggedOut && pathname.includes("/settings/account") ? "" : `?from=${pathname}`
```

Comment:
I have a feeling there may be a cleaner way to determine whether to include the `?from=` path.  I do think it's nice to include here though.

Reply:
I agree that it would be nice for the auth service to remain unconcerned with the routing.

I'm not sure I follow the checking of `user` here - wouldn't it always be `null`? `<Auth>` itself is only rendered if `!user` in [cloudRoutes.tsx](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/packages/cloud/cloudRoutes.tsx#L177)

Reply:
You're right!  I had thought it was needed to handle the redirect upon login, but it is not.  Addressed in https://github.com/airbytehq/airbyte/pull/21175/commits/e196d89d57938ea79252d00247c55d0cb1ab1dc0